ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/MainnetRequestsValidator.java

```diff
@@ -15,33 +15,58 @@
 package org.hyperledger.besu.ethereum.mainnet.requests;
 
 import org.hyperledger.besu.datatypes.RequestType;
+import org.hyperledger.besu.ethereum.core.Request;
 
-public class MainnetRequestsValidator {
-  public static RequestsValidatorCoordinator pragueRequestsValidator(
-      final RequestContractAddresses requestContractAddresses) {
-    return new RequestsValidatorCoordinator.Builder()
-        .addValidator(RequestType.WITHDRAWAL, new WithdrawalRequestValidator())
-        .addValidator(RequestType.CONSOLIDATION, new ConsolidationRequestValidator())
-        .addValidator(
-            RequestType.DEPOSIT,
-            new DepositRequestValidator(requestContractAddresses.getDepositContractAddress()))
-        .build();
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Validates requests within a block against a set of predefined validators. This class delegates
+ * the validation of requests of specific types to corresponding validators. It ensures that
+ * requests are properly ordered, have a valid root, and meet the criteria defined by their
+ * validators.
+ */
+public class MainnetRequestsValidator implements RequestsValidator {
+  private static final Logger LOG = LoggerFactory.getLogger(MainnetRequestsValidator.class);
+
+  /**
+   * Validates a block's requests by ensuring they are correctly ordered, have a valid root, and
+   * pass their respective type-specific validations.
+   *
+   * @param maybeRequests The list of requests to be validated.
+   * @return true if all validations pass; false otherwise.
+   */
+  @Override
+  public boolean validate(final Optional<List<Request>> maybeRequests) {
+    if (maybeRequests.isEmpty()) {
+      LOG.warn("Must contain requests (even if empty list)");
+      return false;
+    }
+
+    if (!isRequestOrderValid(maybeRequests.get())) {
+      LOG.warn("Ordering across requests must be ascending by type");
+      return false;
+    }
+
+    if (!allRequestTypesAreContained(maybeRequests.get())) {
+      LOG.warn("Requests must contain all request types");
+      return false;
+    }
```

Comment:
Will this work if we add a new request type for an futere fork?

Reply:
No it won't work as the list of request types will be different for each fork. This will also need to change for upcoming execution request changes, as there could be empty requests.

Maybe we don't even need this check? Think only the first two validations are actually necessary

Reply:
Removed this size check since we won't be able do this soon with the upcoming changes anyway